dist/src/main/java/io/camunda/application/commons/rest/PackageSpecificJackson2HttpMessageConverter.java

```diff
@@ -22,10 +21,7 @@ public PackageSpecificJackson2HttpMessageConverter(final String basePackage) {
 
   @Override
   public boolean canRead(final Type type, final Class<?> contextClass, final MediaType mediaType) {
-    if (Optional.ofNullable(contextClass)
```

Comment:
contextClass is actually the controller class, while type seems to be what we actually wanna match on, it's also aligned with the logic in canWrite then, which receives the type of the object to serialize.
<img width="798" height="55" alt="image" src="https://github.com/user-attachments/assets/4dfe666c-6899-41ce-966c-1dd8f462c78b" />